File: namoosori_stage3_2/src/javastory/club/stage3/step4/da/map/io/MemoryMap.java
package javastory.club.stage3.step4.da.map.io;

import javastory.club.stage3.step1.entity.board.Posting;
import javastory.club.stage3.step1.entity.board.SocialBoard;
import javastory.club.stage3.step1.entity.club.CommunityMember;
import javastory.club.stage3.step1.entity.club.TravelClub;

import java.util.LinkedHashMap;
import java.util.Map;

public class MemoryMap {

    private static MemoryMap memoryMap;

    private Map<String, CommunityMember> memberMap;
    private Map<String, TravelClub> clubMap;
    private Map<String, SocialBoard> boardMap;
    private Map<String, Posting> postingMap;
    private Map<String, Integer> autoIdMap;

    public MemoryMap() {
        memberMap = new LinkedHashMap<>();
        clubMap = new LinkedHashMap<>();
        boardMap = new LinkedHashMap<>();
        postingMap = new LinkedHashMap<>();
        autoIdMap = new LinkedHashMap<>();
    }

    public static MemoryMap getInstance(){

        if(memoryMap == null){
            memoryMap = new MemoryMap();
        }
        return memoryMap;
    }

    public Map<String , Integer> getAutoIdMap(){
        return this.autoIdMap;
    }

    public Map<String, CommunityMember> getMemberMap() {
        return memberMap;
    }

    public Map<String, TravelClub> getClubMap() {
        return clubMap;
    }

    public Map<String, SocialBoard> getBoardMap() {
        return boardMap;
    }

    public Map<String, Posting> getPostingMap() {
        return postingMap;
    }
}

File: nextree/React+MobX+TypeScript/travelclubtest/src/model/MemoryMap.ts
import TravelClub from "../entity/club/TravelClub";
import CommunityMember from "../entity/club/CommunityMember";
import Posting from "../entity/board/Posting";

let uniqueInstance: any;

class MemoryMap {

    private _clubMap: any;
    private _memberMap : any;
    private _postingMap : any;
    private _autoIdMap : any;

    constructor() {
        if (uniqueInstance) {
            return uniqueInstance;
        }
        this._clubMap = new Map();
        this._memberMap = new Map();
        this._postingMap = new Map();
        this._autoIdMap = new Map();
        uniqueInstance = this; //자기 자신을 참조한다...?
    }

    get clubMap(): Map<string, TravelClub> {
        return this._clubMap;
    }

    get memberMap(): Map<string, CommunityMember>{
        return this._memberMap;
    }

    get postingMap() : Map<string, Posting>{
        return this._postingMap;
    }

    get autoIdMap() : Map<string, number>{
        return this._autoIdMap;
    }
}
export default new MemoryMap();